curvefs/src/client/fuse_client.cpp

```diff
@@ -247,15 +247,54 @@ void FuseClient::FuseOpDestroy(void *userdata) {
     return;
 }
 
-void FuseClient::GetDentryParamFromInode(
-    const std::shared_ptr<InodeWrapper> &inodeWrapper_,
+void GetDentryParamFromInode(
+    const FuseClientOption &option,
+    const std::shared_ptr<InodeWrapper> &inodeWrapper,
     fuse_entry_param *param) {
     memset(param, 0, sizeof(fuse_entry_param));
-    param->ino = inodeWrapper_->GetInodeId();
+    param->ino = inodeWrapper->GetInodeId();
     param->generation = 0;
-    inodeWrapper_->GetInodeAttrLocked(&param->attr);
-    param->attr_timeout = option_.attrTimeOut;
-    param->entry_timeout = option_.entryTimeOut;
+    inodeWrapper->GetInodeAttrLocked(&param->attr);
+    param->attr_timeout = option.attrTimeOut;
+    param->entry_timeout = option.entryTimeOut;
+}
+
+void InodeAttr2ParamAttr(const InodeAttr &inodeAttr, struct stat *attr) {
+    attr->st_ino = inodeAttr.inodeid();
+    attr->st_mode = inodeAttr.mode();
+    attr->st_nlink = inodeAttr.nlink();
+    attr->st_uid = inodeAttr.uid();
+    attr->st_gid = inodeAttr.gid();
+    attr->st_size = inodeAttr.length();
+    attr->st_rdev = inodeAttr.rdev();
+    attr->st_atim.tv_sec = inodeAttr.atime();
+    attr->st_atim.tv_nsec = inodeAttr.atime_ns();
+    attr->st_mtim.tv_sec = inodeAttr.mtime();
+    attr->st_mtim.tv_nsec = inodeAttr.mtime_ns();
+    attr->st_ctim.tv_sec = inodeAttr.ctime();
+    attr->st_ctim.tv_nsec = inodeAttr.ctime_ns();
+    attr->st_blksize = kOptimalIOBlockSize;
+
+    switch (inodeAttr.type()) {
+        case metaserver::TYPE_S3:
+            attr->st_blocks = (inodeAttr.length() + 511) / 512;
+            break;
+        default:
+            attr->st_blocks = 0;
+            break;
+    }
+}
+
+void GetDentryParamFromInode(
```

Comment:
There 2 GetDentryParamFromInode here，maybe one of them can call the other, thus delegating

Reply:
Now there are tow entrances to this function:
1. already have inodeWrapper(create interface)
2. already have inodeAttr(lookup etc.)
The combine of the two functions is achievable, but may have extra transform.

The possible way is delete the related function of get stat attr from inode in inodeWrapper, and instead of get inodeAttr first and transform inodeAttr to stat attr here, so can retain only one 'GetDentryParamFromInode'.

Reply:
fixed